presentation/src/main/java/team/aliens/dms_android/feature/notice/NoticesViewModel.kt

```diff
@@ -0,0 +1,179 @@
+) : MviViewModel<NoticesUiState, NoticesUiEvent>(NoticesUiState.initial()) {
+
+    init {
+        fetchNotices()
+    }
+
+    override fun updateState(event: NoticesUiEvent) {
+        when (event) {
+            is NoticesUiEvent.SetOrder -> {
+                setOrder()
+                fetchNotices()
+            }
+
+            is NoticesUiEvent.FetchNoticeDetails -> {
+                setNoticeId(event.noticeId)
+                fetchNoticeDetails()
+            }
+
+            is NoticesUiEvent.FetchWhetherNewNoticesExist -> {
+                fetchWhetherNewNoticesExist()
+            }
+        }
+    }
+
+    private fun fetchNotices() {
+        viewModelScope.launch(Dispatchers.IO) {
+            kotlin.runCatching {
+                fetchNoticesUseCase(
+                    fetchNoticesInput = FetchNoticesInput(
+                        order = uiState.value.order,
+                    ),
+                )
+            }.onSuccess {
+                setNotices(
+                    notices = it.notices.toModel()
+                )
+            }.onFailure {
+
+            }
+        }
+    }
+
+    private fun fetchNoticeDetails() {
+        viewModelScope.launch(Dispatchers.IO) {
+            kotlin.runCatching {
+                fetchNoticeDetailsUseCase(
+                    fetchNoticeDetailsInput = FetchNoticeDetailsInput(
+                        noticeId = uiState.value.noticeId!!,
+                    ),
+                )
+            }.onSuccess {
+                setNoticeDetails(
+                    noticeDetails = it.toModel()
+                )
+            }.onFailure {
+
+            }
+        }
+    }
+
+    internal fun fetchWhetherNewNoticesExist() {
+        viewModelScope.launch {
+            kotlin.runCatching {
+                fetchWhetherNewNoticesExistUseCase()
+            }.onSuccess {
+                setWhetherNewNoticesExists(
+                    whetherNewNoticesExists = it.newNotices,
+                )
+            }.onFailure {
+
+            }
+        }
+    }
+
+    private fun setNotices(
+        notices: List<Notice>
+    ) {
+        setState(
+            newState = uiState.value.copy(
+                notices = notices,
+            )
+        )
+    }
+
+    private fun setNoticeDetails(
+        noticeDetails: Notice,
+    ) {
+        setState(
+            newState = uiState.value.copy(
+                noticeDetails = noticeDetails,
+            )
+        )
+    }
+
+    private fun setWhetherNewNoticesExists(
+        whetherNewNoticesExists: Boolean,
+    ) {
+        setState(
+            newState = uiState.value.copy(
+                whetherNewNoticesExists = whetherNewNoticesExists,
+            )
+        )
+    }
+
+    private fun setOrder() {
+        setState(
+            newState = uiState.value.copy(
+                order = when (uiState.value.order) {
+                    Order.NEW -> Order.OLD
+                    else -> Order.NEW
+                }
```

Comment:
뭐 크게 상관 없다고 생각되지만, 케이스가 OLD, NEW 두 개만 있어서 `if-else`로 구현할 수 있을 것 같네요
```suggestion
                order = {
                    val currentOrder = uiState.value.order
                    if (currentOrder == Order.NEW) Order.OLD else Order.NEW
                }
```